lib/massmailer.js

```diff
@@ -45,22 +45,25 @@ exports.sendAll = async (req, res) => {
     const bodies = [];
 
     for (const application of applications) {
-        const typeAndOrder = application.participant_type
-            ? (application.participant_type + ' (' + application.participant_order + ')')
-            : 'not set';
-        const text = req.body.text
-            .replace(/\{first_name\}/ig, application.first_name)
-            .replace(/\{last_name\}/ig, application.last_name)
-            .replace(/\{participant_type_order\}/ig, typeAndOrder)
-            .replace(/\{body_name\}/ig, application.body_name);
-
-        // Using the custom mailer template, it accepts only body as a parameter
-        // and sends the body as it was passed.
-        const notificationEmail = mails.find((m) => application.user_id === m.id).notification_email;
-        to.push(notificationEmail);
-        bodies.push({ body: text });
-
-        logger.info({ notificationEmail }, 'Prepared email');
+        const user = mails.find((m) => application.user_id === m.id);
+        if (user !== undefined) {
```

Comment:
1) you can use something like

```
if (!user) {
   continue
}
```

and avoid 1 level of nesting

2) is this the common case when this happens? at least it's better to add warn/error logging here in case this happens

Reply:
I've added the continue and a warn logging. Not sure what you mean with common case, but currently the massmailer does not work if one of the users it's trying to send the email to is deleted from core so this works around that.

Reply:
I mean, why such things happen in the first place, is it normal?

Reply:
It can happen. Chair wanted to send massmailer to all participants of the Spring Agora but in the meantime one of the users got deleted in core so it could not retrieve that notification email. That caused it to fail the entire function and no emails could be sent. Luckily I knew which user that was so we could change the application status of that user so it would be filtered out but with this change the massmailer would just ignore that user


Later on we should figure out more on what should happen if an user is deleted from core; which data we can keep anonymized and such

Reply:
ah, okay, so deleted users then. That makes sense in that case.